File: Php/framework/framework/io/smartpath.php
<?php

namespace Smartkernel\Framework\IO {

    use Smartkernel\Framework\IO\SmartFile;
    use Smartkernel\Framework\Computer\SmartComputer;

/*
     * 路径操作
     */

    class SmartPath {
        /*
         * 获得文件路径信息（pathinfo函数不能正确处理多字节编码类型的路径名）
         * 
         * @param string $filePath，文件路径
         * @return array，结果
         * 
         */

        public static function getPathInfo($filePath) {

            $result = array();
            $result ["dirname"] = rtrim(substr($filePath, 0, strrpos($filePath, SmartComputer::DirectorySeparator)), SmartComputer::DirectorySeparator) . SmartComputer::DirectorySeparator;
            $result ["basename"] = ltrim(substr($filePath, strrpos($filePath, SmartComputer::DirectorySeparator)), SmartComputer::DirectorySeparator);
            $result ["extension"] = substr(strrchr($filePath, "."), 1);
            $result ["filename"] = ltrim(substr($result ["basename"], 0, strrpos($result ["basename"], ".")), SmartComputer::DirectorySeparator);
            return $result;
        }

        //目录分隔符号
        const DirectorySeparatorChar = SmartComputer::DirectorySeparator;

        /*
         * 获得目录名称
         * 
         * @param string $filePath，路径
         * @return string，结果
         * 
         */

        public static function getDirectoryName($filePath) {
            return self::getPathInfo($filePath)["dirname"];
        }

        /*
         * 获得扩展名称
         * 
         * @param string $filePath，路径
         * @return string，结果
         * 
         */

        public static function getExtension($filePath) {
            return self::getPathInfo($filePath)["extension"];
        }

        /*
         * 获得文件名称（不包含扩展名）
         * 
         * @param string $filePath，路径
         * @return string，结果
         * 
         */

        public static function getFileNameWithoutExtension($filePath) {
            return self::getPathInfo($filePath)["filename"];
        }

        /*
         * 获得文件名称（包含扩展名）
         * 
         * @param string $filePath，路径
         * @return string，结果
         * 
         */

        public static function getFileName($filePath) {
            $pathInfo = self::getPathInfo($filePath);
            return $pathInfo["filename"] . "." . $pathInfo["extension"];
        }

        /*
         * 获得文件根目录
         * 
         * @param string $filePath，路径
         * @return string，结果
         * 
         */

        public static function getPathRoot($filePath) {
            $filePath = SmartFile::getFormatFilePath($filePath);
            $parts = explode(self::DirectorySeparatorChar, $filePath);
            return $parts[0] . self::DirectorySeparatorChar;
        }

    }

}

File: Framework/Java/framework/framework/smartkernel/framework/io/SmartPath.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework.io;

import java.io.File;

import smartkernel.framework.*;
import smartkernel.framework.computer.*;

/**
 * 路径
 * 
 */
public class SmartPath {
	/**
	 * 获得扩展名
	 * 
	 * @param filePath
	 *            ，路径
	 * @return，结果
	 */
	public static String getExtension(String filePath) {
		filePath = SmartFile.getFormatFilePath(filePath);
		String[] filePathParts = filePath.split("\\"
				+ SmartComputer.DirectorySeparator);
		String filename = filePathParts[filePathParts.length - 1];
		String[] nameSplit = filename.split("[.]");
		return nameSplit[nameSplit.length - 1];
	}

	/**
	 * 获得文件名（不包含扩展名）
	 * 
	 * @param filePath
	 *            ，路径
	 * @return，结果
	 */
	public static String getFileNameWithoutExtension(String filePath) {
		filePath = SmartFile.getFormatFilePath(filePath);
		String[] filePathParts = filePath.split("\\"
				+ SmartComputer.DirectorySeparator);
		String filename = filePathParts[filePathParts.length - 1];
		String[] nameSplit = filename.split("[.]");
		return SmartString.trimEnd(filename, "."
				+ nameSplit[nameSplit.length - 1]);
	}

	/**
	 * 获得文件名
	 * 
	 * @param filePath
	 *            ，路径
	 * @return，结果
	 */
	public static String getFileName(String filePath) {
		filePath = SmartFile.getFormatFilePath(filePath);
		String[] filePathParts = filePath.split("\\"
				+ SmartComputer.DirectorySeparator);
		String filename = filePathParts[filePathParts.length - 1];
		return filename;
	}

	/**
	 * 获得文件夹路径（有结尾分隔符号）
	 * 
	 * @param filePath
	 *            ，文件路径
	 * @return，结果
	 */
	public static String getDirectoryName(String filePath) {
		filePath = SmartFile.getFormatFilePath(filePath);
		String parent = new File(filePath).getParent();
		parent = parent == null ? "" : parent;
		return parent + SmartComputer.DirectorySeparator;
	}
}
